lxd/container.go

```diff
@@ -748,22 +748,24 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 
 func containerConfigureInternal(c container) error {
 	// Find the root device
-	for _, m := range c.ExpandedDevices() {
-		if m["type"] != "disk" || m["path"] != "/" || m["size"] == "" {
-			continue
-		}
+	localDevices := c.LocalDevices()
+	_, rootDiskDevice := containerGetRootDiskDevice(localDevices)
+	if rootDiskDevice["pool"] == "" {
+		expandedDevices := c.ExpandedDevices()
+		_, rootDiskDevice = containerGetRootDiskDevice(expandedDevices)
+	}
```

Comment:
Right, you can indeed end up in that case because we don't want to prevent profile config changes because of container local configuration.

But such a container then becomes invalid and won't start as it'll fail the rootfs check then (more than one path=/).